valence-rs/valence
# [ECS] re-implement opening inventories

This adds an `OpenInventory` component that indicates that an entity has opened and is viewing an inventory.

This doesn't yet implement actually allowing clients to modify the inventories. 

This makes it possible to query all entities for open inventories. We also don't have to manage a `_modified` flag, we can just use the tools bevy_ecs already provides. It also helps to avoid cluttering up the `Client` component more than it already is.
`window_id` is intensionally placed on `Client` because it's state that pertains to that specific connection, not the inventory that is currently open.

### Test Plan

1. Run the example.

```bash
cargo run --example inventory_test
```

2. Connect to the server.
3. An inventory will be shown to you when you finish joining.
4. Close the inventory to cycle to the next inventory.
5. All inventories should have a diamond pickaxe in slot 0.
